### PR TITLE
Audit: make sure service commands exist

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
And here we have 1 of the 2 reasons that I wanted to make the service block.

**Comment by SMillerDev** [comment]
Hmm, I need to somehow make this audit require an installed formula. Any suggestions?

**Comment by MikeMcQuaid** [comment]
> Hmm, I need to somehow make this audit require an installed formula. Any suggestions?

https://github.com/Homebrew/brew/blob/a4ea7fb279b581fa0e7783f2b0d76f6a7ed6372e/Library/Homebrew/formula_cellar_checks.rb#L278-L297

**Comment by SMillerDev** [comment]
It seems kinda weird to add it to the Cellar checks since it checks the formula, not the cellar. But I guess that can work until someone has a better idea.

**Comment by Bo98** [comment]
> It seems kinda weird to add it to the Cellar checks since it checks the formula, not the cellar. But I guess that can work until someone has a better idea.

You can think of it as checking the Cellar that the file referenced by the service block exists.

**Review by Rylan12 — DISMISSED** [review]
Looks good to me. One question if you don't mind: why/when/how would you specify multiple `run` commands?

**Comment by SMillerDev** [comment]
> Looks good to me. One question if you don't mind: why/when/how would you specify multiple run commands?

It's multiple parts, e.g. `[bin/'httpd', '-D', 'foreground']`

**Comment by Rylan12** [comment]
Ah gotcha, thanks!

File: Library/Homebrew/formula_cellar_checks.rb [hunk 1]
```diff
@@ -275,12 +275,22 @@ def check_python_symlinks(name, keg_only)
     "Python formulae that are keg-only should not create `pip3` and `wheel3` symlinks."
   end
 
+  def check_service_command(formula)
+    return unless formula.prefix.directory?
+    return unless formula.service?
+
+    return "Service command undefined" if formula.service.command.empty?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_cellar_checks.rb:282]:
```suggestion
    return "Service command blank" if formula.service.command.blank?
```

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**